src/vivarium_gates_mncnh/components/pregnancy.py

```diff
@@ -0,0 +1,176 @@
+import pandas as pd
+from vivarium import Component
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+from vivarium_public_health.utilities import get_lookup_columns
+
+from vivarium_gates_mncnh.components.children import NewChildren
+from vivarium_gates_mncnh.constants import data_keys
+from vivarium_gates_mncnh.constants.data_values import DURATIONS, PREGNANCY_OUTCOMES
+from vivarium_gates_mncnh.constants.metadata import ARTIFACT_INDEX_COLUMNS
+
+
+class Pregnancy(Component):
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def columns_created(self):
+        return [
+            "pregnancy_outcome",
+            "pregnancy_duration",
+            "sex_of_child",
+            "birth_weight",
+            "gestational_age",  # Why repeated?
+        ]
+
+    @property
+    def sub_components(self):
+        return super().sub_components + [self.new_children]
+
+    @property
+    def initialization_requirements(self) -> dict[str, list[str]]:
+        return {
+            "requires_columns": [],
+            "requires_values": ["birth_outcome_probabilities"],
+            "requires_streams": [],
+        }
+
+    def __init__(self):
+        super().__init__()
+        self.new_children = NewChildren()
+
+    def setup(self, builder: Builder):
+        """Performs this component's simulation setup.
+
+        Parameters
+        ----------
+        builder : `engine.Builder`
+            Interface to several simulation tools.
+        """
+        super().setup(builder)
```

Comment:
Nit: you don't need this `super()` call. Its super-class is `Component`, whose `setup()` method is empty.